# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: modules/libpref/Preferences.cpp
# Commit: 3d57d5bbd058
# Full Hash: 3d57d5bbd0580aa3f258bc6c06a009545ba72889
# Author: Daniel Holbert <dholbert@cs.stanford.edu>
# Date: 2023-04-04 09:39:15
# Description:
#   Bug 1826234: Fix GetObserverService error-handling check in Preferences::InitInitialObjects. r=xpcom-reviewers,emilio
#   
#   In mozilla-central before this patch, we had a boilerplate
#   NS_ENSURE_SUCCESS(rv, rv) after our call to GetObserverService()
#   
# ==============================================================================

diff -r 9f05d67ae4ee -r 3d57d5bbd058 modules/libpref/Preferences.cpp
--- a/modules/libpref/Preferences.cpp	Tue Apr 04 00:31:47 2023 +0000
+++ b/modules/libpref/Preferences.cpp	Tue Apr 04 01:06:01 2023 +0000
@@ -5025,7 +5025,9 @@
                                 NS_PREFSERVICE_APPDEFAULTS_TOPIC_ID);
 
   nsCOMPtr<nsIObserverService> observerService = services::GetObserverService();
-  NS_ENSURE_SUCCESS(rv, rv);
+  if (NS_WARN_IF(!observerService)) {
+    return NS_ERROR_FAILURE;
+  }
 
   observerService->NotifyObservers(nullptr, NS_PREFSERVICE_APPDEFAULTS_TOPIC_ID,
                                    nullptr);
